app/controllers/hosts_controller.rb

```diff
@@ -414,6 +415,34 @@ def multiple_destroy
   def multiple_build
   end
 
+  def rebuild_config
+  end
+
+  def submit_rebuild_config
+    all_fails = {:tftp => [], :dns => [], :dhcp => []}
+
+    @hosts.each do |host|
+      result = host.rebuild_config
+      result.each_pair do |k, v|
+        all_fails[k] << host.name unless v
+      end
+    end
+
+    message = ''
+    all_fails.each_pair do |key, values|
+      unless values.empty?
+        message << (_("%{r} rebuild failed for hosts: %{s}. ") % { :s => values.join(', '), :r => key.to_s.upcase})
```

Comment:
a) Use more meaningful variables ('r', 's') in translated strings as translators need to understand the context.
b) For hostnames you may want to use `values.to_sentence` which gives a localised string, such as "foo, bar and baz".
c) Remove the whitespace at the end of the string - if it must be there, put it outside of the translation.